opslevel/resource_opslevel_service.go

```diff
@@ -1,5 +1,333 @@
+				Computed: true,
+			},
+			"lifecycle_alias": schema.StringAttribute{
+				Description: "The lifecycle stage of the service.",
+				Optional:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The display name of the service.",
+				Required:    true,
+			},
+			"owner": schema.StringAttribute{
+				Description: "The team that owns the service. ID or Alias my be used.",
+				Optional:    true,
+			},
+			"owner_id": schema.StringAttribute{
+				Description: "The team ID that owns the service.",
+				Optional:    true,
+			},
+			"preferred_api_document_source": schema.StringAttribute{
+				Description: "The API document source (PUSH or PULL) used to determine the displayed document. If null, we use the order push and then pull.",
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(opslevel.AllApiDocumentSourceEnum...),
+				},
+			},
+			"product": schema.StringAttribute{
+				Description: "A product is an application that your end user interacts with. Multiple services can work together to power a single product.",
+				Optional:    true,
+			},
+			"tags": schema.ListAttribute{
+				ElementType: types.StringType,
+				Description: "A list of tags applied to the service.",
+				Optional:    true,
+			},
+			"tier_alias": schema.StringAttribute{
+				Description: "The software tier that the service belongs to.",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func (r *ServiceResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data ServiceResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	serviceCreateInput := opslevel.ServiceCreateInput{
+		Description:    data.Description.ValueStringPointer(),
+		Framework:      data.Framework.ValueStringPointer(),
+		Language:       data.Language.ValueStringPointer(),
+		LifecycleAlias: data.LifecycleAlias.ValueStringPointer(),
+		Name:           data.Name.ValueString(),
+		Product:        data.Product.ValueStringPointer(),
+		TierAlias:      data.TierAlias.ValueStringPointer(),
+	}
+
+	if data.Owner.ValueString() != "" {
+		serviceCreateInput.OwnerInput = opslevel.NewIdentifier(data.Owner.ValueString())
+	}
+
+	service, err := r.client.CreateService(serviceCreateInput)
+	if err != nil || service == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to create service, got error: %s", err))
+		return
+	}
+
+	givenAliases, diags := ListValueToStringSlice(ctx, data.Aliases)
+	if diags != nil && diags.HasError() {
+		resp.Diagnostics.AddError("Config error", fmt.Sprintf("Unable to handle given service aliases: '%s'", data.Aliases))
+		return
+	}
+	err = reconcileServiceAliases(*r.client, givenAliases, service)
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to create service aliases: '%s'", givenAliases))
```

Comment:
```suggestion
		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to reconcile service aliases: '%s'", givenAliases))
```

Error can be returned on delete failure as well.